addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SqlInjectionScanRule.java

```diff
@@ -1913,11 +1956,16 @@ private void expressionBasedAttack(
                                 confirmExpressionOutputUnstripped,
                                 originalParam,
                                 modifiedParamValueConfirm);
+                String confirmExpressionLocationHeader =
+                        msg.getResponseHeader().getHeader(HttpHeader.LOCATION);
```

Comment:
Could extract a convenience method for this accepting HttpMessage and returning String.

Then leverage it throughout.